playbooks/dalite.yml

```diff
@@ -0,0 +1,24 @@
+---
+# Playbook for setting up Dalite.
+# See https://github.com/open-craft/dalite-ng
+
+- name: Set up Dalite servers
+  hosts: dalite
+  become: true
+  roles:
+    - name: common-server
+      TARSNAP_KEY: "{{ DALITE_LOGROTATE_TARSNAP_KEY }}"
+      TARSNAP_KEY_REMOTE_LOCATION: "/root/tarsnap-logrotate.key"
+      TARSNAP_BACKUP_PRE_SCRIPT: "/usr/local/sbin/dalite-pre-backup.sh"
+      TARSNAP_BACKUP_SCRIPT_LOCATION: "{{ DALITE_TARSNAP_BACKUP_SCRIPT }}"
+      TARSNAP_ARCHIVE_NAME: "dalite-logs"
+      TARSNAP_CACHE: "/var/cache/tarsnap-logrotate"
+      TARSNAP_BACKUP_FOLDERS: "{{ DALITE_LOG_DOWNLOAD_LOG_DIR }} {{ DALITE_LOG_DOWNLOAD_DB_DIR }}"
+      TARSNAP_BACKUP_SNITCH: "{{ DALITE_LOG_TARSNAP_SNITCH }}"
+      # Backup of logs is initiated by logrotate, which incidentally
+      # also creates data to be backed up.
+      TARSNAP_CRONTAB_STATE: "absent"
+    - forward-server-mail
```

Comment:
I looked into porting to the latest version of Ansible.  Ansible 2.5 was released a month ago, and now has official Python 3 support.  If we migrate, we can use the `import_role` and `import_playbook` modules, which will make some of the semantics of including roles less weird (like the way in which tags and variables are inherited).  Once you have finished and merged this PR, I will give porting to Ansible 2.5.1 a shot – we will also be able to remove the Python 2 installation task.  And I also can't wait to merge all the other roles into this repo as well – I just wish I had done it before last week.

Reply:
OK that sounds like a plan. Note that I may just merge all other roles right here before I want to merge this PR. It's quite mechanical and boring but I can probably write a script and just get it done quickly.

Reply:
I can also merge the other roles, since I have some time available, and I would like to get this PR merged as soon as possible, so I can base any future work on the new version.  Here's what I propose:

* Third-party roles to be inlcuded as git submodules:
  * geerlingguy.security
  * kamaln7.swapfile
  * pmbauer.tarsnap  (Side note: I want to get rid of that role – Tarsnap can now be installed via apt)
  * greendayonfire.mongodb
  * geerlingguy.mysql
  * geerlingguy.postgresql
* OpenCraft roles to be included with git subtree:
  * load-balancer – we have CircleCI set up for this role, so we should keep the repo, but reduce the overhead by using subtree instead of submodule
* OpenCraft roles to be included directly:
  * backup-to-tarsnap
  * common-server
  * configure-apt
  * sanity-checker
  * forward-server-mail
  * backup-swift-container
  * dalite
  * rabbitmq
  * elasticsearch
  * postfix
  * mattermost
  * crafty
  * link-shortener
  * accounting
  * mysql
  * postgres
  * mongodb

I could prepare a PR against your branch if you want me to.

Reply:
OK, I'll just go with submodules. I've also just gone ahead and done it now.